File: Ruben - Modulo 2/prueba/src/Punto.java
public class Punto {
    private double x, y;
    public Punto() { x = y = 0; }
    public Punto(double a, double b) { x = a; y = b; }
    public double abscisa() { return x; }
    public double ordenada() { return y; }
    public void abscisa(double a){ x = a; }
    public void ordenada(double b){ y = b; }
    public void trasladar(double a, double b) {
        x += a; y += b;
    }
    public double distancia(Punto pto) {
        return Math.sqrt(Math.pow(x - pto.x, 2) +
                Math.pow(y - pto.y, 2));
    }

    @Override
    public String toString() {
        return "(" + x + ", " + y +")";
    }
}

File: Ruben - Modulo 2/prPuntov1/src/puntosv1/Punto.kt
package puntosv1;
import kotlin.math.pow import kotlin.math.sqrt
class Punto constructor (a: Double, b: Double) {
    private var x: Double;
    private var y: Double;

    init {
        x = a
        y = b
    }

    constructor() : this(0.0, 0.0) {}

    fun getX(): Double = x
    fun getY(): Double = y
    fun distancia(p: Punto): Double =
            sqrt((p.x - this.x).pow(2.0) + (p.y - this.y).pow(2.0))

    fun trasladar(a: Double, b: Double) {
        x += a
        y += b
    }

    override fun toString(): String = "($x,$y)"
}